src/ProductConstructionService/ProductConstructionService.SubscriptionTriggerer/SubscriptionTriggerer.cs

```diff
@@ -26,8 +26,23 @@ public SubscriptionTriggerer(
         _workItemProducerFactory = workItemProducerFactory;
     }
 
-    public async Task CheckSubscriptionsAsync(UpdateFrequency targetUpdateFrequency)
+    public async Task TriggerSubscriptionsAsync(UpdateFrequency targetUpdateFrequency)
     {
+        var workItemProducer = _workItemProducerFactory.CreateProducer<UpdateSubscriptionWorkItem>();
+        foreach (var updateSubscriptionWorkItem in await GetSubscriptionsToTrigger(targetUpdateFrequency))
+        {
+            // TODO https://github.com/dotnet/arcade-services/issues/3811 add some kind of feature switch to trigger specific subscriptions
+            //await workItemProducer.ProduceWorkItemAsync(updateSubscriptionWorkItem);
+            _logger.LogInformation("Queued update for subscription '{subscriptionId}' with build '{buildId}'",
+                    updateSubscriptionWorkItem.SubscriptionId,
+                    updateSubscriptionWorkItem.BuildId);
+        }
+    }
+
+    public async Task<List<UpdateSubscriptionWorkItem>> GetSubscriptionsToTrigger(UpdateFrequency targetUpdateFrequency)
```

Comment:
Is this method here only for the sake of tests?

Reply:
I am not a fan of broadening the API just for the sake of the tests.
We should test this properly which means we should probably uncomment the producer part and let it produce, but only inject a no-op producer. Later, when we want subscriptions triggered, we can remove this no-op producer registration.

In the tests, we can give it a mock producer and verify that it created a queue message per each subscription we wanted.

Reply:
yeah that's true. I think I'll still keep the separated methods, just make that one private